SPDX-FileCopyrightText: 2023 fosslinux <fosslinux@aussies.space>
SPDX-FileCopyrightText: 2022 Thomas Schwinge <thomas@codesourcery.com>  

SPDX-License-Identifier: GPL-3.0-or-later

Backport of commit 25861cf3a88a07c8dca3fb32d098c0ad756bbe38

====
With that, we may then run plain 'autoreconf' for all of GCC's subpackages,
instead of for some of those (that don't use Automake) manually having to run
the applicable combination of 'aclocal', 'autoconf', 'autoheader'.

See also 'AC_CONFIG_MACRO_DIRS'/'AC_CONFIG_MACRO_DIR' usage elsewhere.

diff --git gcc/configure.ac gcc/configure.ac
index e1ef2ecf026..45bf7560e6f 100644
--- gcc/configure.ac
+++ gcc/configure.ac
@@ -25,6 +25,7 @@
 
 AC_INIT
 AC_CONFIG_SRCDIR(tree.cc)
+AC_CONFIG_MACRO_DIRS([../config] [..])
 AC_CONFIG_HEADER(auto-host.h:config.in)
 
 gcc_version=`cat $srcdir/BASE-VER`
diff --git libobjc/configure libobjc/configure
index a8fdc643349..6da20b8e4ff 100755
diff --git libobjc/configure.ac libobjc/configure.ac
index f8f577cfbef..6f58a45d4cb 100644
--- libobjc/configure.ac
+++ libobjc/configure.ac
@@ -20,6 +20,7 @@
 
 AC_INIT(package-unused, version-unused,, libobjc)
 AC_CONFIG_SRCDIR([objc/objc.h])
+AC_CONFIG_MACRO_DIRS([../config] [..])
 GCC_TOPLEV_SUBDIRS
 
 # We need the following definitions because AC_PROG_LIBTOOL relies on them
-- 
2.31.1

